pkg/controller/persistentvolume/persistentvolume_index.go

```diff
@@ -117,16 +92,20 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 				continue
 			}
 
-			if claim.Name == volume.Spec.ClaimRef.Name && claim.Namespace == volume.Spec.ClaimRef.Namespace && claim.UID == volume.Spec.ClaimRef.UID {
-				// exact match! No search required.
+			if isVolumeBoundToClaim(volume, claim) {
```

Comment:
> I think this func should also handle the case of a pvc pre-binding to a pv via spec.VolumeName, 

This function checks only one direction, i.e. whether the **volume** is bound. Claims with `claim.Spec.VolumeName` set should not call `findBestMatchForClaim`/`listByAccessModes` at all (and it's an error if they do).